opacus/tests/grad_samples/conv2d_test.py

```diff
@@ -85,20 +85,22 @@ def test_conv2d(
             rtol=10e-4,
             ew_compatible=is_ew_compatible,
         )
-        # Test 'convolution as a backward' GSM
-        conv2d_gsm = GradSampleModule.GRAD_SAMPLERS[nn.Conv2d]
-        GradSampleModule.GRAD_SAMPLERS[
-            nn.Conv2d
-        ] = convolution2d_backward_as_a_convolution
-        self.run_test(
-            x,
-            conv,
-            batch_first=True,
-            atol=10e-5,
-            rtol=10e-4,
-            ew_compatible=is_ew_compatible,
-        )
-        GradSampleModule.GRAD_SAMPLERS[nn.Conv2d] = conv2d_gsm
+
+        if padding != "same":
```

Comment:
Can you add a comment here explaining why this check is needed? Essentially copy-pasta the PR description/summary here (and may be also in the actual grad_sampler file)